File: c4di_coffee_web/web/models/reading.ex
defmodule C4diCoffeeWeb.Reading do
  use C4diCoffeeWeb.Web, :model

  schema "readings" do
    field :reading_kg, :float
    field :time, Ecto.DateTime
    field :avg_since, Ecto.DateTime
    field :pot_name, :string

    timestamps
  end

  @required_fields ~w(reading_kg time avg_since pot_name)
  @optional_fields ~w()

  @doc """
  Creates a changeset based on the `model` and `params`.

  If no params are provided, an invalid changeset is returned
  with no validation performed.
  """
  def changeset(model, params \\ :empty) do
    model
    |> cast(params, @required_fields, @optional_fields)
  end
end

File: mqtt-client/src/app/Reading.ts
class Reading {
    reading_kg: number;
    time: Date;
    avg_since: Date;
    pot_name: string;

    constructor(reading_kg: number, time: Date, avg_since: Date, pot_name: string){
    	this.reading_kg = reading_kg;
    	this.time = time;
    	this.avg_since = avg_since;
    	this.pot_name = pot_name;
    }

}